portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,41 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<String> messages = new ArrayList<>();
+  private String json;
+
+  /** Converts Messages ArrayList to Json */
+  public void toJson() throws IOException {
+    json = new Gson().toJson(messages);
```

Comment:
It might be better here to return a String of the json directly, rather than storing it in a member variable. This approach adds a sort of "state" and requires the json member variable to be in-sync, but you could alternatively return the json directly that way it's always the up-to-date version.

Reply:
Sounds like a good idea! Thank you Mark.